src/core/cm/launcher/nodemanager.hpp

```diff
@@ -0,0 +1,156 @@
+        resourcemanager::ResourceManagerItf& resourceManager, storagestate::StorageStateItf& storageState,
+        InstanceRunnerItf& runner);
+
+    /**
+     * Starts node manager.
+     *
+     * @return Error.
+     */
+    Error Start();
+
+    /**
+     * Stops node manager.
+     *
+     * @return Error.
+     */
+    Error Stop();
+
+    /**
+     * Updates running instances of a single node.
+     *
+     * @param nodeID node identifier.
+     * @param statuses instance status list.
+     * @return Error.
+     */
+    Error UpdateNodeInstances(const String& nodeID, const Array<InstanceStatus>& statuses);
+
+    /**
+     * Updates instances for all nodes.
+     *
+     * @param instances instances.
+     * @return Error.
+     */
+    Error UpdateNodeInstances(const Array<SharedPtr<Instance>>& instances);
+
+    /**
+     * Returns nodes ordered by priorities.
+     *
+     * @param nodes output array of nodes.
+     * @return Error.
+     */
+    Error GetNodesByPriorities(Array<Node*>& nodes);
+
+    /**
+     * Finds node by identifier.
+     *
+     * @param nodeID node identifier.
+     * @return Node*.
+     */
+    Node* FindNode(const String& nodeID);
+
+    /**
+     * Returns nodes.
+     *
+     * @return Array<Node>&.
+     */
+    Array<Node>& GetNodes();
+
+    /**
+     * Sets up state storage for instance.
+     *
+     * @param nodeConfig node configuration.
+     * @param serviceConfig service configuration.
+     * @param gid group identifier.
+     * @param info instance information.
+     * @return Error.
+     */
+    Error SetupStateStorage(
+        const NodeConfig& nodeConfig, const oci::ServiceConfig& serviceConfig, gid_t gid, aos::InstanceInfo& info);
+
+    /**
+     * Checks whether instance is running.
+     *
+     * @param instance instance identifier.
+     * @return bool.
+     */
+    bool IsRunning(const InstanceIdent& instance);
+
+    /**
+     * Checks whether instance is scheduled.
+     *
+     * @param instance instance identifier.
+     * @return bool.
+     */
+    bool IsScheduled(const InstanceIdent& instance);
+
+private:
+    static constexpr auto cDefaultResourceRation = 50.0;
+
+    /**
+     * Handler of node info change event.
+     */
+    void OnNodeInfoChanged(const UnitNodeInfo& info) override;
+
+    uint64_t GetReqStateSize(const NodeConfig& nodeConfig, const oci::ServiceConfig& serviceConfig) const;
+    uint64_t GetReqStorageSize(const NodeConfig& nodeConfig, const oci::ServiceConfig& serviceConfig) const;
+
+    uint64_t ClampResource(uint64_t value, const Optional<uint64_t>& quota) const;
+    uint64_t GetReqStateFromNodeConfig(
+        const Optional<uint64_t>& quota, const Optional<ResourceRatios>& nodeRatios) const;
+    uint64_t GetReqStorageFromNodeConfig(
+        const Optional<uint64_t>& quota, const Optional<ResourceRatios>& nodeRatios) const;
+
+    nodeinfoprovider::NodeInfoProviderItf* mNodeInfoProvider    = nullptr;
```

Comment:
please unify initialization for members 
```suggestion
    nodeinfoprovider::NodeInfoProviderItf* mNodeInfoProvider    {};
```